connectors/kafka/src/main/java/io/camunda/connector/kafka/model/KafkaPropertiesUtil.java

```diff
@@ -0,0 +1,114 @@
+      "org.apache.kafka.common.serialization.StringSerializer";
+  private static final String BYTE_ARRAY_SERIALIZER =
+      "org.apache.kafka.common.serialization.ByteArraySerializer";
+
+  // Kafka client is built using java.utils.Properties.
+  // This method creates properties required to establish connection and produce messages.
+  public static Properties assembleKafkaClientProperties(KafkaConnectorRequest request) {
+    Properties props = new Properties();
+
+    // Step 1: collect properties directly from the form
+    if (request.authentication() != null) {
+      Properties authProps =
+          KafkaPropertiesUtil.produceAuthenticationProperties(request.authentication());
+      props.putAll(authProps);
+    }
+    Properties topicProps = new Properties();
+    topicProps.setProperty(
+        ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, request.topic().bootstrapServers());
+    props.putAll(topicProps);
+
+    if (request.message() != null) { // can be valid in case of inbound
+      Properties messageProps = new Properties();
+      messageProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, STRING_SERIALIZER);
+      if (request.avro() == null) {
+        messageProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, STRING_SERIALIZER);
+      } else {
+        messageProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, BYTE_ARRAY_SERIALIZER);
+      }
+      props.putAll(messageProps);
+    }
+
+    // Step 2: set default recommended properties
+    // See the list of possible values at org.apache.kafka.clients.producer.ProducerConfig
+    props.put(CommonClientConfigs.SESSION_TIMEOUT_MS_CONFIG, SESSION_TIMEOUT_MS_RECOMMENDED_VALUE);
+    props.put(ProducerConfig.CLIENT_DNS_LOOKUP_CONFIG, CLIENT_DNS_LOOKUP_RECOMMENDED_VALUE);
+    props.put(ProducerConfig.ACKS_CONFIG, ACKS_RECOMMENDED_VALUE);
+    props.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, DELIVERY_TIMEOUT_MS_RECOMMENDED_VALUE);
+    props.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, MAX_BLOCK_MS_RECOMMENDED_VALUE);
+    props.put(ProducerConfig.LINGER_MS_CONFIG, LINGER_MS);
+    props.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, REQUEST_TIMEOUT_MS);
+    props.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, HEARTBEAT_INTERVAL_MS);
+    props.put(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG, DEFAULT_API_TIMEOUT_MS);
+
+    // Step 3: override properties provided by the client
+    if (request.additionalProperties() != null && !request.additionalProperties().isEmpty()) {
+      props.putAll(request.additionalProperties());
+    }
+
+    return props;
+  }
+
+  public static Properties produceAuthenticationProperties(KafkaAuthentication authentication) {
+    Properties authProps = new Properties();
+
+    // Both username and password arrived empty thus not setting security config.
+    if ((authentication.username() == null || authentication.username().isBlank())
+        && (authentication.password() == null || authentication.password().isBlank())) {
```

Comment:
If there's an Apache Commons `StringUtils` is in deps, you could simplify to:
```suggestion
    if (StringUtils.isBlank(authentication.username()) && StringUtils.isBlank(authentication.username())) {
```